components/odh-notebook-controller/controllers/notebook_webhook.go

```diff
@@ -369,6 +403,57 @@ func (w *NotebookWebhook) maybeRestartRunningNotebook(ctx context.Context, req a
 	return mutatedNotebook, &UpdatesPending{Reason: diff}, nil
 }
 
+func InjectProxyConfigEnvVars(notebook *nbv1.Notebook) error {
```

Comment:
There should be an envtest for this. When we decide to get the PR in, this is the first thing I'd like to look into.

We'd need to add a fake Proxy CRD, and have some test that starts Notebook with and without a proxy being configured. We don't much care for a scenario when Proxy CRD is missing (that's what we're testing now, actually, the `return false` when `err != nil` in `ClusterWideProxyIsEnabled` above) because OpenShift always has Proxy CRD.

Reply:
feel free to push to my fork, if you want to. I am fine with that in my case.

Reply:
@jiridanek can you make a follow-up Story in Jira for this, please? Enhancements to code. 
I assume @harshad16  took this aspect into account when merging in as-is.
What I can say for certain: in daily operations, the controller is running just fine.